### PR TITLE
Lf table upgrade

### PR DESCRIPTION
**Pull request by LucaFumis**
This PR updates the cancer biomarker widget to the new table component, greatly simplifying from the previous custom table.
As before, all the data is batch loaded (see note below), but now using the useBatchDownloader hook for consistency.

The cancer biomarkers API endpoint does allow for pagination, so a server side approach on the table would leverage that. However the endpoint doesn't support search, so to my understanding table filtering (specifically requested in the ticket) would not be possible with server side approach. This could be changed/improved in the future.

The whole `custom/filterTable.js` component is redundant and could be removed: it contains examples of the old summary charts, but I imagine that these would be implemented differently now if so required.

Closes https://github.com/opentargets/platform/issues/1122

### CODE REVIEW

File: src/public/target/sections/CancerBiomarkers/Section.js [hunk 1]
```diff
@@ -1,9 +1,93 @@
...
+  );
+
+  useEffect(() => {
+    getAllBiomarkers().then(d => {
+      setRows(d);
+    });
+  }, []);
+
+  return (
+    <Table
```
Comment (mirandaio) [inline src/public/target/sections/CancerBiomarkers/Section.js:82]: The cancer biomarkers API provides server side paging: ` cancerBiomarkers(page: Pagination): CancerBiomarkers`
We could take advantage of this server side paging by trying to only the get the necessary rows needed for a particular page. Here's a section that takes advantage of this server side paging: https://github.com/opentargets/platform-app/blob/alpha/src/public/target/sections/RelatedTargets/Section.js

Comment (LucaFumis) [reply src/public/target/sections/CancerBiomarkers/Section.js:82]: Thanks @mirandaio I agree that server side pagination in this case is a good idea.
However, to my understanding that would not work with table search/filtering: 
the API endpoint `cancerBiomarkers(page: Pagination)` takes *only* the pagination parameter, unlike the `knownDrugs` that also takes care of table filtering:
```knownDrugs(
freeTextQuery: String
size: Int
cursor: [String!]
): KnownDrugs```

I don't mind either way - @andrewhercules what do you think: search or load/pagination performance?

Comment (andrewhercules) [reply src/public/target/sections/CancerBiomarkers/Section.js:82]: The [Cancer Biomarkers dataset](https://www.cancergenomeinterpreter.org/biomarkers) is quite small - it does not need to be server-side pagination. I believe the target with the most rows is [ERBB2](https://alpha.targetvalidation.org/target/ENSG00000141736) with 104. 

To keep most data tables consistent, I would ignore the server-side pagination and allow for search. But if there's a performance or code issue, then I defer to your expertise and we can implement the server-side pagination instead.

**Comment by LucaFumis** [comment]
Converting the PR to draft as there seem to be some bugs - testing with ERBB2 doesn't work as I was expecting, so I need to revise the loading of data.

File: src/public/target/sections/CancerBiomarkers/Section.js (hunk 1 above)
Comment (javfg) [reply src/public/target/sections/CancerBiomarkers/Section.js:82]: @LucaFumis you are right, you cannot use searching/filtering on a server side mode table if there is no support in the backend, as you don't have all the rows on the client. And it looks like CancerBiomarkers does not provide anything besides pagination.

As Andrew said, it looks like there is not a lot of rows in those queries, so you can keep the client side table implementation.

Comment (mirandaio) [reply src/public/target/sections/CancerBiomarkers/Section.js:82]: If backend is not planning to provide a `freeTextQuery` like in known drugs, then yeah, in that case, client side paging in the only option.

**Comment by LucaFumis** [comment]
Ready for review, however the latest commit [dcfeb01](https://github.com/opentargets/platform-app/commit/dcfeb016e23a3e0a5e2ae95a68cb53c8e29248ff) is not showing in the PR. 
Github was a little flakey this morning, so perhaps that might be the cause.

**Pull request merged**